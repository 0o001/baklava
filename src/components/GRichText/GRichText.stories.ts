import { storiesOf } from '@storybook/vue';
import GRichText from './';

const stories = storiesOf('GRichText', module);

stories
  .add('Default', () => ({
    components: {
      GRichText,
    },
    data () {
      return {
        data: 'default text',
        data2: 'default text',
        data3: 'default text',
        opts: {
          placeholder: '',
          modules: {
            toolbar: [
              ['bold', 'italic', 'underline', 'strike'],
              ['blockquote', 'code-block'],
              [{ 'header': 1 }, { 'header': 2 }],
              [{ 'list': 'ordered' }, { 'list': 'bullet' }],
              [{ 'script': 'sub' }, { 'script': 'super' }],
              [{ 'indent': '-1' }, { 'indent': '+1' }],
              [{ 'direction': 'rtl' }],
              [{ 'size': ['small', false, 'large', 'huge'] }],
              [{ 'header': [1, 2, 3, 4, 5, 6, false] }],
              [{ 'font': [] }],
              [{ 'color': [] }, { 'background': [] }],
              [{ 'align': [] }],
              ['clean'],
              ['link', 'image', 'video']
            ],
          }
        }
      }
    },
    template: `<div style="margin: 40px 200px 200px">
        <g-rich-text v-model="data" feedback="Buraya yalnizca 50 karakter girebilirsiniz!"></g-rich-text>
        <br />
        <g-rich-text v-model="data2" :error="true" feedback="Lutfen gerekli alanlari doldurun!" :opts="opts"></g-rich-text>
        <br />
        <g-rich-text v-model="data3" :success="true" feedback="Aferin cok iyi girdin yaziyi!" :opts="opts"></g-rich-text>
        <br />
        <g-rich-text :disable="true" feedback="Su an disabled, buraya yazi yazamazsin!" :opts="opts"></g-rich-text>
    </div>`
  }))

