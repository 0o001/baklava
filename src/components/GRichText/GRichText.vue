<template>
  <div :class="wrapperClass">
    <quill-editor
      :options="options"
      :value="value"
      :disabled="disable"
      v-on="$listeners"
    />
    <div class="feedback">
      {{ feedback }}
    </div>
  </div>
</template>
<script>
import { quillEditor } from 'vue-quill-editor';

export default {
  name: 'GRichText',
  components: {
    quillEditor,
  },
  props: {
    value: {
      type: String,
      default: '',
    },
    feedback: {
      type: String,
      default: '',
    },
    success: {
      default: false,
      type: Boolean,
    },
    error: {
      default: false,
      type: Boolean,
    },
    disable: {
      default: false,
      type: Boolean,
    },
    options: {
      type: Object,
      default: () => ({}),
    },
    opts: {
      type: Object,
      default: () => ({}),
    },
  },
  data () {
    return {
      defaultOpts: {
        placeholder: '', // placeholder'li test
        modules: {
          toolbar: [
            ['bold', 'italic', 'underline', 'strike'],
            [{ size: ['small', false, 'large', 'huge'] }],
            [{ list: 'ordered' }, { list: 'bullet' }],
            [{ direction: 'rtl' }],
            [{ align: [] }],
          ],
        },
      },
    };
  },
  beforeMount () {
    Object.assign(this.options, this.defaultOpts, this.opts);
  },
  computed: {
    wrapperClass () {
      return {
        'g-rich-text': true,
        '-error': this.error,
        '-valid': this.success,
        '-disable': this.disable,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
  @import '~quill/dist/quill.core.css';
  @import '~quill/dist/quill.snow.css';
  @import '~quill/dist/quill.bubble.css';

  .g-rich-text {
    .quill-editor {
      ::v-deep .ql-toolbar {
        border: solid 1px #afbbca;
        border-radius: 4px 4px 0 0;
        padding: 4px;
      }

      ::v-deep .ql-container {
        border: solid 1px #afbbca;
        border-radius: 0 0 4px 4px;
        border-top: 0;
      }

      ::v-deep .ql-editor strong {
        font-weight: bold;
      }

      ::v-deep .ql-editor em {
        font-style: italic;
      }
    }

    .feedback {
      font-family: var(--font-family-base);
      font-size: var(--font-size-12);
      font-weight: normal;
      margin-top: 5px;
      margin-left: 10px;
      color: var(--dark-grey-500);
    }

    &.-error {
      ::v-deep .ql-container {
        border-color: var(--red-500) !important;
      }

      ::v-deep .ql-toolbar {
        border-color: var(--red-500) !important;
      }

      .feedback {
        color:  var(--red-500) !important;
      }
    }
    &.-valid {
      ::v-deep .ql-container {
        border-color: var(--green-500) !important;
      }

      ::v-deep .ql-toolbar {
        border-color: var(--green-500) !important;
      }

      .feedback {
        color:  var(--green-500) !important;
      }
    }
    &.-disable {
      ::v-deep .ql-container {
        border-color: var(--mid-grey-800) !important;
        background: var(--bg-grey-500);
      }

      ::v-deep .ql-toolbar {
        border-color: var(--mid-grey-800) !important;
      }

      .feedback {
        color: var(--dark-grey-500) !important;
      }
    }
  }

</style>
