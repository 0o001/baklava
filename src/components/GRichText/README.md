# GRichText

## Props

| Name       | Type      | Description | Default      |
| ---------- | --------- | ----------- | ------------ |
| `v-model`  | `String`  |             | `""`         |
| `feedback` | `String`  |             | `""`         |
| `success`  | `Boolean` |             | `false`      |
| `error`    | `Boolean` |             | `false`      |
| `disable`  | `Boolean` |             | `false`      |
| `options`  | `Object`  |             | `() => ({})` |
| `opts`     | `Object`  |             | `() => ({})` |

## Data

| Name          | Type     | Description | Initial value                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
| ------------- | -------- | ----------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `defaultOpts` | `object` |             | `{"placeholder":{"type":"string","value":"","raw":"\"\""},"modules":{"type":"object","value":{"toolbar":{"type":"array","value":"[ ['bold', 'italic', 'underline', 'strike'], [{ size: ['small', false, 'large', 'huge'] }], [{ list: 'ordered' }, { list: 'bullet' }], [{ direction: 'rtl' }], [{ align: [] }], ]","raw":"[ ['bold', 'italic', 'underline', 'strike'], [{ size: ['small', false, 'large', 'huge'] }], [{ list: 'ordered' }, { list: 'bullet' }], [{ direction: 'rtl' }], [{ align: [] }], ]"}},"raw":"{\"toolbar\":{\"type\":\"array\",\"value\":\"[ ['bold', 'italic', 'underline', 'strike'], [{ size: ['small', false, 'large', 'huge'] }], [{ list: 'ordered' }, { list: 'bullet' }], [{ direction: 'rtl' }], [{ align: [] }], ]\",\"raw\":\"[ ['bold', 'italic', 'underline', 'strike'], [{ size: ['small', false, 'large', 'huge'] }], [{ list: 'ordered' }, { list: 'bullet' }], [{ direction: 'rtl' }], [{ align: [] }], ]\"}}"}}` |

## Computed Properties

| Name           | Type     | Description                                     |
| -------------- | -------- | ----------------------------------------------- |
| `wrapperClass` | `object` | **Dependencies:** `error`, `success`, `disable` |

## Events

| Name    | Description                                               |
| ------- | --------------------------------------------------------- |
| `input` | <br/>**Arguments**<br/><ul><li>**`event: any`**</li></ul> |

## Methods

### onInput()

**Syntax**

```typescript
onInput(event: unknown): void
```

