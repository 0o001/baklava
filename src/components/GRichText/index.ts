import GRichText from './GRichText.vue';

export default GRichText;
