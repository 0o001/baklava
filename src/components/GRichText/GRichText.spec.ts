import { shallowMount } from '@vue/test-utils';
import GRichText from './';


describe('GRichText', () => {
  let wrapper: any;

  const renderWrapper = (options: any = {}) => {
    wrapper = shallowMount(GRichText, {
      ...options,
    });
  };

  it('should props with default values ​​when no props are set', () => {
    renderWrapper();
    expect(wrapper.vm.wrapperClass).toEqual({
      'g-rich-text': true,
      '-error': false,
      '-valid': false,
      '-disable': false,
    });
  });

  it('should have valid class ​​when success prop is set to true', () => {
    renderWrapper({
      propsData: {
        success: true,
      },
    });

    expect(wrapper.vm.wrapperClass).toEqual({
      'g-rich-text': true,
      '-error': false,
      '-valid': true,
      '-disable': false,
    });
  });

  it('should have error class ​​when error prop is set to true', () => {
    renderWrapper({
      propsData: {
        error: true,
      },
    });

    expect(wrapper.vm.wrapperClass).toEqual({
      'g-rich-text': true,
      '-error': true,
      '-valid': false,
      '-disable': false,
    });
  });

  it('should have disable class ​​when disable prop is set to true', () => {
    renderWrapper({
      propsData: {
        disable: true,
      },
    });

    expect(wrapper.vm.wrapperClass).toEqual({
      'g-rich-text': true,
      '-error': false,
      '-valid': false,
      '-disable': true,
    });
  });

  it('should have correct default options', () => {
    renderWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should emit input event when onInput is called', () => {
    const event = {
      type: 'test',
    };

    wrapper.vm.onInput(event);

    expect(wrapper.emitted().input.length).toEqual(1);
  });



});
