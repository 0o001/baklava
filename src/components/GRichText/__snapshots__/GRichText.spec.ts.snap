// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GRichText should have correct default options 1`] = `
<div
  class="g-rich-text"
>
  <quill-editor-stub
    globaloptions="[object Object]"
    options="[object Object]"
    value=""
  />
   
  <div
    class="feedback"
  >
    
    
  
  </div>
</div>
`;
